src/CabalHelper/Runtime/Main.hs

```diff
@@ -643,29 +743,42 @@ componentEntrypoints (CLib Library {..})
     = ChLibEntrypoint
         (map gmModuleName exposedModules)
         (map gmModuleName $ otherModules libBuildInfo)
+#if CH_MIN_VERSION_Cabal(2,0,0)
+        (map gmModuleName signatures)
+#else
+        [] -- no signatures prior to Cabal 2.0
+#endif
 #if CH_MIN_VERSION_Cabal(2,0,0)
 componentEntrypoints (CFLib (ForeignLib{..}))
     = ChLibEntrypoint
         []
         (map gmModuleName $ otherModules foreignLibBuildInfo)
+        []
 #endif
 componentEntrypoints (CExe Executable {..})
-    = ChExeEntrypoint modulePath (map gmModuleName $ otherModules buildInfo)
+    = ChExeEntrypoint
+#if CH_MIN_VERSION_Cabal(2,0,0)
+        ( head ((hsSourceDirs buildInfo) ++ ["."]) </> modulePath)
```

Comment:
What on earth is going on here? It looks like you're changing the semantics of the field which is a really bad idea. If not at least explain why this has to be different from the Cabal<2 case in a comment.

Reply:
It would not load in the GhcSessionTest without doing this, and it is what the straight `cabal build` does